src/controllers/executionPlanWebviewController.ts

```diff
@@ -0,0 +1,157 @@
+    // @ts-ignore
+    private xmlPlanFileName: string
+  ) {
+    super(
+      context,
+      `${xmlPlanFileName} ${LocalizedConstants.executionPlan}`,  // Sets the webview title
+      WebviewRoute.executionPlanDocument,
+      {},
+      vscode.ViewColumn.Active,
+      {
+        dark: vscode.Uri.joinPath(
+          context.extensionUri,
+          "media",
+          "executionPlan_inverse.svg"
+        ),
+        light: vscode.Uri.joinPath(
+          context.extensionUri,
+          "media",
+          "executionPlan.svg"
+        ),
+      }
+    );
+    this.state.isLoading = true;
+    this.initialize();
+  }
+
+  private async initialize() {
+    this.state.sqlPlanContent = this.executionPlanContents;
+	  this.state.theme = vscode.window.activeColorTheme.kind === vscode.ColorThemeKind.Dark ? "dark" : "light";
+    this.state.localizedConstants = LocalizedConstants;
+
+    await this.createExecutionPlanGraphs();
+    this.state.totalCost = this.calculateTotalCost();
+    this.registerRpcHandlers();
+    this.state.isLoading = false;
+  }
+
+  private registerRpcHandlers() {
+    this.registerReducer("getExecutionPlan", async (state, payload) => {
+      await this.createExecutionPlanGraphs();
+
+      return {
+        ...state,
+        executionPlan: this.state.executionPlan,
+        executionPlanGraphs: this.state.executionPlanGraphs
+      };
+    });
+    this.registerReducer("saveExecutionPlan", async (state, payload) => {
+      const homeDir = homedir();
+      const documentsFolder = vscode.Uri.file(`${homeDir}/Documents`);
+
+      let filename: vscode.Uri;
+      let counter = 1;
+      if (await fileExists(documentsFolder, `plan.sqlplan`)) {
+        while (
+          await fileExists(documentsFolder, `plan${counter}.sqlplan`)
+        ) {
+          counter += 1;
+        }
+        filename = vscode.Uri.joinPath(
+          documentsFolder,
+          `plan${counter}.sqlplan`
+        );
+      } else {
+        filename = vscode.Uri.joinPath(documentsFolder, "plan.sqlplan");
+      }
+
+      // Show a save dialog to the user
+      const saveUri = await vscode.window.showSaveDialog({
+        defaultUri: filename,
+        filters: {
+          "SQL Plan Files": ["sqlplan"],
+        },
+      });
+
+      if (saveUri) {
+        // Write the content to the new file
+        await vscode.workspace.fs.writeFile(
+          saveUri,
+          Buffer.from(payload.sqlPlanContent)
+        );
+      }
+
+      return state;
+    });
+    this.registerReducer("showPlanXml", async (state, payload) => {
+      const planXmlDoc = await vscode.workspace.openTextDocument({
+        content: payload.sqlPlanContent,
+        language: 'xml'
+      });
+
+      await vscode.window.showTextDocument(planXmlDoc);
+
+      return state;
+    });
+    this.registerReducer("showQuery", async (state, payload) => {
+      const sqlDoc = await vscode.workspace.openTextDocument({

```

Comment:
Can you use the existing service to open sql documents?

For example:
https://github.com/microsoft/vscode-mssql/blob/4c4f13274f0881a613af0c822fd0b0d49fcc2b2d/src/tableDesigner/tableDesignerWebViewController.ts#L215